nucypher/blockchain/eth/actors.py

```diff
@@ -518,11 +533,22 @@ def deploy_beneficiary_contracts(self,
     @staticmethod
     def __read_allocation_data(filepath: str) -> list:
         with open(filepath, 'r') as allocation_file:
-            data = allocation_file.read()
-            try:
-                allocation_data = json.loads(data)
-            except JSONDecodeError:
-                raise
+            if filepath.endswith(".csv"):
+                allocation_data = list()
+                reader = csv.DictReader(allocation_file)
+                for entry in reader:
+                    entry['amount'], entry['duration_seconds'] = int(entry['amount']), int(entry['duration_seconds'])
+                    allocation_data.append(entry)
+            else:
+                data = allocation_file.read()
+                try:
+                    allocation_data = json.loads(data)
+                except JSONDecodeError:
```

Comment:
Hm, and ignore any other errors? I understand that this `try` block came from the original version, but perhaps we can just remove it - the CSV part doesn't have anything like that.

Reply:
Current catch block doesn't achieve anything, so yes, perhaps it's better to simply remove it.

Reply:
✅